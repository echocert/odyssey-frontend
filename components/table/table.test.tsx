import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";
import TableView from "./table";
describe("Table Component", () => {
  it("renders the Table Component with Fake Data", () => {
    render(
      <TableView
        tableHeader={[{ text: "Test", key: "test", type: "test" }]}
        tableData={[
          {
            test: "test",
          },
        ]}
      />
    );
  });

  it("Verify the snapshot of the Table Component with Fake Data is matched", () => {
    const wrap = create(
      <TableView
        tableHeader={[{ text: "Test", key: "test", type: "test" }]}
        tableData={[
          {
            test: "test",
          },
        ]}
      />
    ).toJSON();
    expect(wrap).toMatchSnapshot();
  });
});
