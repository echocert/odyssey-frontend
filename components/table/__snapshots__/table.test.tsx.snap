// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Verify the snapshot of the Table Component with Fake Data is matched 1`] = `
<div
  className="table-wrapper"
>
  <div
    className="table-header-wrapper"
  >
    <div
      style={
        {
          "flex": 1,
        }
      }
    />
    <div />
  </div>
  <table
    className="table-view responsiveTable"
  >
    <thead
      data-testid="thead"
    >
      <tr
        className="table-view__row"
        data-testid="tr"
      >
        <th
          className="table-view__header"
          data-testid="th"
        >
          <label
            className="label-hero-l2 label-header-l2 label-l2 label-header"
            style={{}}
          >
            No
          </label>
        </th>
        <th
          className="table-view__header"
          data-field-type="test"
          data-sort-key="test"
          data-testid="th"
          onClick={[Function]}
        >
          <label
            className="label-hero-l2 label-header-l2 label-l2 label-header"
            style={{}}
          >
            Test
          </label>
        </th>
        <th
          className="table-view__header"
          data-field-type="none"
          data-sort-key="view"
          data-testid="th"
          onClick={[Function]}
        >
          <label
            className="label-hero-l2 label-header-l2 label-l2 label-header"
            style={{}}
          >
            View Icon
          </label>
        </th>
      </tr>
    </thead>
    <tbody
      data-testid="tbody"
    >
      <tr
        className="table-view__row"
        data-testid="tr"
      >
        <td
          className="table-data pivoted"
          data-testid="td"
        >
          <div
            className="tdBefore"
            data-testid="td-before"
          >
            <label
              className="label-hero-l2 label-header-l2 label-l2 label-header"
              style={{}}
            >
              No
            </label>
          </div>
          1
        </td>
        <td
          className="table-data pivoted"
          data-testid="td"
        >
          <div
            className="tdBefore"
            data-testid="td-before"
          >
            <label
              className="label-hero-l2 label-header-l2 label-l2 label-header"
              style={{}}
            >
              Test
            </label>
          </div>
           
          <label
            className="label-subheader-l3 label-l3 label-subheader"
            style={{}}
          >
            test
          </label>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
